lib/datadog/appsec/processor.rb

```diff
@@ -16,9 +16,38 @@ def self.===(other)
         end
       end
 
+      # Context wraps libddwaf's context
+      class Context
+        attr_reader :time, :time_ext, :timeouts
+
+        def initialize(processor)
+          @context = Datadog::AppSec::WAF::Context.new(processor.send(:handle))
```

Comment:
This is a weird break in the abstraction... Have you considered passing the `handle` directly when creating the new `Context`?

Reply:
You mean using `send` here? I was annoyed by that too and considered it.

The abstraction goal here with `Processor` and `Processor::Context` is to hide away `Datadog::AppSec::WAF`, so exposing the handle (which is a `Datadog::AppSec::WAF::Handle`) runs against that.

Reply:
Ah, no, I meant actually passing in the fully-built context from the `Processor`. E.g. instead of

```ruby
      def new_context
        Context.new(self)
      end
```

why not do

```ruby
      def new_context
        Context.new(Datadog::AppSec::WAF::Context.new(@handle))
      end
```

...that way the handle never needs to go out of the processor, and the `Processor::Context` doesn't actually need to care about where the object it wraps comes from.

Reply:
Hmm, I understand. Then `Datadog::AppSec::WAF::Context` becomes part of the public interface of `Processor::Context` as an argument to `new`. I feel like that would make it a leaky abstraction.

Reply:
Fair enough... I guess?

I think having to use the `send(:handle)` here is already breaking the abstraction anyway but seeing as this already quite localized, and it's rather small, let's keep it as-is, and we can always revisit it later as the codebase evolves. Marking as resolved 😄